forms-flow-api/src/formsflow_api/services/external/keycloak.py

```diff
@@ -153,5 +153,61 @@ def update_request(  # pylint: disable=inconsistent-return-statements
             current_app.logger.debug(f"Keycloak response: {response}")
         except Exception as err_code:
             raise f"Request to Keycloak Admin APIs failed., {err_code}"
+        response.raise_for_status()
         if response.status_code == 204:
             return f"Updated - {url_path}"
+
+    def get_groups(self):
+        """Return groups."""
+        current_app.logger.debug("Getting groups")
+        group_list_response = self.get_request(
+            url_path="groups?briefRepresentation=false"
+        )
+        current_app.logger.debug("Groups %s", group_list_response)
+        return group_list_response
+
+    def get_roles(self, search: str = ""):
+        """Return roles."""
+        current_app.logger.debug("Getting roles")
+        client_id = self.get_client_id()
+        roles = self.get_request(f"clients/{client_id}/roles?search={search}")
+        current_app.logger.debug("Client roles %s", roles)
+        return roles
+
+    @profiletime
+    def delete_request(self, url_path):
```

Comment:
Keycloak admin DELETE APIs accept the body, so better to add `data` parameter to the request